python/lsst/meas/astrom/fitSipDistortion.py

```diff
@@ -0,0 +1,420 @@
+        doc="order of SIP polynomial",
+        dtype=int,
+        default=4,
+        min=0,
+    )
+    numRejIter = lsst.pex.config.RangeField(
+        doc="number of rejection iterations",
+        dtype=int,
+        default=3,
+        min=0,
+    )
+    rejSigma = lsst.pex.config.RangeField(
+        doc="Number of standard deviations for clipping level",
+        dtype=float,
+        default=3.0,
+        min=0.0,
+    )
+    nClipMin = lsst.pex.config.Field(
+        doc="Minimum number of matches to reject when sigma-clipping",
+        dtype=int,
+        default=0
+    )
+    nClipMax = lsst.pex.config.Field(
+        doc="Maximum number of matches to reject when sigma-clipping",
+        dtype=int,
+        default=1
+    )
+    maxScatterArcsec = lsst.pex.config.RangeField(
+        doc="maximum median scatter of a WCS fit beyond which the fit fails (arcsec); " +
+        "be generous, as this is only intended to catch catastrophic failures",
+        dtype=float,
+        default=10,
+        min=0,
+    )
+    refUncertainty = lsst.pex.config.Field(
+        doc="RMS uncertainty in reference catalog positions, in pixels.  Will be added " +
+        "in quadrature with measured uncertainties in the fit.",
+        dtype=float,
+        default=0.25,
+    )
+    nGridX = lsst.pex.config.Field(
+        doc="Number of X grid points used to invert the SIP reverse transform.",
+        dtype=int,
+        default=100,
+    )
+    nGridY = lsst.pex.config.Field(
+        doc="Number of Y grid points used to invert the SIP reverse transform.",
+        dtype=int,
+        default=100,
+    )
+    gridBorder = lsst.pex.config.Field(
+        doc=("When setting the gird region, how much to extend the image "
+             "bounding box (in pixels) before transforming it to intermediate "
+             "world coordinates using the initial WCS."),
+        dtype=float,
+        default=50.0,
+    )
+
+
+class FitSipDistortionTask(lsst.pipe.base.Task):
+    """Fit a TAN-SIP WCS given a list of reference object/source matches
+
+    FitSipDistortionTask is a drop-in replacement for
+    :py:class:`lsst.meas.astrom.FitTanSinWcsTask`.  It is built on fundamentally
+    stronger fitting algorithms, but has received significantly less testing.
+
+    Like :py:class:`lsst.meas.astrom.FitTanSinWcsTask`, this task is most
+    easily used as the wcsFitter component of
+    :py:class:`lsst.meas.astrom.AstrometryTask`; it can be enabled in a config
+    file via e.g.
+
+    .. code-block:: py
+
+       from lsst.meas.astrom import FitSipDistortionTask
+       config.(...).astometry.wcsFitter.retarget(FitSipDistortionTask)
+
+    Algorithm
+    ---------
+
+    The algorithm used by FitSipDistortionTask involves three steps:
+
+     - We set the CRVAL and CRPIX reference points to the mean position, while
+       holding the CD matrix fixed to its input value.  This work is done by
+       the makeInitialWcs method.
```

Comment:
I thought this was a summary, though I see that I just said "mean position" when I should have said "mean position of all the matches".  Does that clarify things?

I think we could add a config option to leave CRVAL and/or CRPIX fixed at the positions from the given initial WCS, but this matches the behavior of `FitTanSipWcsTask` and I didn't see anything wrong with that.